pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/bloom/OnHeapGuavaBloomFilterCreator.java

```diff
@@ -47,9 +48,18 @@ public class OnHeapGuavaBloomFilterCreator implements BloomFilterCreator {
 
   private final File _bloomFilterFile;
   private final BloomFilter<String> _bloomFilter;
+  private final FieldSpec.DataType _dataType;
 
+  // TODO: This method is here for compatibility reasons, should be removed in future PRs
```

Comment:
nit: In my experience, it is better to define an "exit criteria" for the deprecated feature. Meaning "when" in the future can this be safely removed.  In my previous team, we used something like:
```
exit_criteria =  "when PR #XYZ is merged in oss"
```

In this case, you can mention when it will no longer matter. It can also be as straightforward as "remove in releaseX.Y.Z"
I don't know if Pinot has a standard practice for removing deprecated feature/code. But just my 2 cents.